src/installer.ts

```diff
@@ -147,8 +147,10 @@ export async function installKamel(context: vscode.ExtensionContext): Promise<Er
 		try {
 			kamelUrl = await versionUtils.getDownloadURLForCamelKTag(versionToUse, platform);
 		} catch (error) {
-			extension.shareMessageInMainOutputChannel(error);
-			throw new Error(error);
+			if (error instanceof Error) {
+				extension.shareMessageInMainOutputChannel(error.message);
+				throw new Error(error.message);
+			}
```

Comment:
I answer this question in reverse order:

> based on the link you provided, they are providing this kind of snippet code but at first sight it seems weird. 

Yes, I think so too, firstly I would like to make just a small cast to String(), but it is official documentation and I decided to make it that way

> Do you have more information to it? have you seen this practice elsewhere? Is there a specific reason to do it this way?

More information you can find here in issue -> https://github.com/microsoft/TypeScript/issues/36775

> In case it is not an Error, it will be swallowed and we won't have any traces. 

Yes, you are absolutely right here :disappointed: 
WDYT to combine official snippet with the cast to string? 
```
let message = ""
if (error instanceof Error) {
	message = error.message;
} else {
	message = String(error);				
}
extension.shareMessageInMainOutputChannel(message);
throw new Error(message);			   
```
or
```
let message = error instanceof Error ? error.message : String(error);
extension.shareMessageInMainOutputChannel(message);
throw new Error(message);
```

Reply:
combining official snippet with the cast to string seems a good alternative.

For the compilation distinction, I think I understand from where it comes from. I think it is better to create a specific issue to discuss it separately. The priority is lower.